error.go

```diff
@@ -4,29 +4,36 @@ import (
 	"fmt"
 )
 
-func New(msg string) *Kind {
-	return &Kind{Message: msg}
-}
-
+// Kind represents a kind of an error, from a Kind you can generate as many
+// Error instances of you want of this Kind
 type Kind struct {
 	Message string
 }
 
+// New returns a Kind with the given msg
+func New(msg string) *Kind {
+	return &Kind{Message: msg}
+}
+
+// New returns a new Error, values can be pass to it if the Kind was created
```

Comment:
I'd say "New returns a new Error of this Kind. The Kind message will be formatted with the given values in case the message had a printf-like format."